### PR TITLE
Add ffwd metrics client

### PR DESCRIPTION
**Pull request by smstone**
**What this PR does / why we need it**:
This is needed so we can wrap Gordon Janitor plugins with a metrics object.

**Which issue(s) this PR fixes**

Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note

```

@spotify/alf

### CODE REVIEW

**Comment by ocsw** [comment]
A lot of this looks copied from gordon core?  I was hoping that we could just import most of that from the core package.

**Comment by smstone** [comment]
@ocsw  yeah it is, just feels weird i suppose to import from `gordon` though. i can take that approach so there is less copy pasta

**Comment by smstone** [comment]
@spotify/alf i've imported the metrics modules from `gordon`. can you PTAL?

File: tests/unit/test_plugins_loader.py [hunk 1]
```diff
@@ -187,39 +208,31 @@ def test_get_namespaced_config(namespace, exp_config, plugins, loaded_config):
         loaded_config, namespace, all_plugins)
 
     assert exp_config == ret_config
-    assert namespace == ret_namespace
+#     assert namespace == ret_namespace
```
Comment (ocsw) [inline tests/unit/test_plugins_loader.py]: Can this be removed?

File: tests/unit/test_plugins_loader.py [hunk 2]
```diff
@@ -267,14 +280,78 @@ def test_load_plugins_exceptions(plugins, exp_inited_plugins, loaded_config,
...
+
+def test_get_metrics_returns_plugin(metrics_mock, plugins_incl_metrics):
+    """MetricRelay should load if both implements interface and configured."""
+    config = {'core': {'metrics': 'mock-provider-name'}}
+    actual = plugins_loader._get_metrics_plugin(config, plugins_incl_metrics)
+    assert isinstance(actual, MetricRelayStub)
+
+
+def test_get_metrics_not_installed_raises(installed_plugins):
+    """Return None if config or name incorrect."""
```
Comment (ocsw) [inline tests/unit/test_plugins_loader.py]: This should say raises.

File: tests/unit/test_plugins_loader.py [hunk 3]
```diff
@@ -267,14 +280,78 @@ def test_load_plugins_exceptions(plugins, exp_inited_plugins, loaded_config,
...
+
+def test_get_metrics_returns_ffwd(loaded_config, plugins_incl_metrics):
+    loaded_config['core'].update({'metrics': 'ffwd'})
+    actual = plugins_loader._get_metrics_plugin(
+        loaded_config, plugins_incl_metrics)
+    assert isinstance(actual, ffwd.SimpleFfwdRelay)
+
+
+def test_get_metrics_returns_plugin(metrics_mock, plugins_incl_metrics):
+    """MetricRelay should load if both implements interface and configured."""
```
Comment (ocsw) [inline tests/unit/test_plugins_loader.py]: I'm not clear on 'implements interface' - do you mean 'has an entry point'?

**Comment by ocsw** [comment]
LGTM aside from some very small things.  I checked, though - the entire plugins_loader.py is almost identical between core and janitor now.  I was kinda hoping to import that too.

File: gordon_janitor/__init__.py [hunk 4]
```diff
@@ -15,7 +15,7 @@
 # limitations under the License.
 
 __author__ = 'Lynn Root'
-__version__ = '0.0.1.dev4'
+__version__ = '0.0.1.dev5'
```
Comment (ocsw) [inline gordon_janitor/__init__.py]: This should be in a separate release PR.

**Comment by smstone** [comment]
@ocsw imported that too now, not sure why i didn't think of that

File: tests/unit/conftest.py [hunk 5]
```diff
@@ -17,8 +17,24 @@
...
+import asyncio
+import os
+
 import pkg_resources
 import pytest
+import zope.interface
+
+from gordon_janitor import interfaces
+
+PLUGIN_NAMES = [
```
Comment (ocsw) [inline tests/unit/conftest.py]: How much of these changes do we still need?

File: tests/unit/fixtures/test-gordon-janitor.toml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+# NOTICE - Test Fixture for Gordon Janitor
```
Comment (ocsw) [inline tests/unit/fixtures/test-gordon-janitor.toml:1]: Do we still need this?

File: gordon_janitor/plugins_loader.py [hunk 7]
```diff
@@ -1,210 +0,0 @@
...
-"""
-
-import logging
-
-import pkg_resources
-
-from gordon_janitor import exceptions
-
-
-PLUGIN_NAMESPACE = 'gordon_janitor.plugins'
```
Comment (ocsw) [inline gordon_janitor/plugins_loader.py:64]: You'll still need this.

**Review by ocsw — CHANGES_REQUESTED** [review]

File: tests/unit/fixtures/test-gordon-janitor.toml (hunk 6 above)
Comment (smstone) [reply tests/unit/fixtures/test-gordon-janitor.toml:1]: yeah it's used `conftest.py` then used in `test_main` to ensure we can load a test config

File: gordon_janitor/plugins_loader.py [hunk 8]
```diff
@@ -54,157 +54,4 @@
...
 """
 
-import logging
-
-import pkg_resources
-
-from gordon_janitor import exceptions
-
-
 PLUGIN_NAMESPACE = 'gordon_janitor.plugins'
```
Comment (ocsw) [inline gordon_janitor/plugins_loader.py]: Er, to clarify, you'll need to set it in the imported module, because I'm pretty sure it's only at module scope.

Comment (ocsw) [reply gordon_janitor/plugins_loader.py]: from gordon import plugins_loader
plugins.loader.PLUGIN_NAMESPACE = 'gordon_janitor.plugins'

Comment (smstone) [reply gordon_janitor/plugins_loader.py]: got it!

**Review by ocsw — APPROVED** [review]

**Comment by ocsw** [comment]
+1

**Pull request merged**